modules/light-clients/07-tendermint/misbehaviour_handle.go

```diff
@@ -53,6 +53,29 @@ func (cs ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCode
 	case *Misbehaviour:
 		// The correctness of Misbehaviour ClientMessage types is ensured by calling VerifyClientMessage prior to this function
 		// Thus, here we can return true, as ClientMessage is of type Misbehaviour
+		// if heights are equal check that this is valid misbehaviour of a fork
+		// otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
+		if msg.Header1.GetHeight().EQ(msg.Header2.GetHeight()) {
+			blockID1, err := tmtypes.BlockIDFromProto(&msg.Header1.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
+
+			blockID2, err := tmtypes.BlockIDFromProto(&msg.Header2.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
+
+			// Ensure that Commit Hashes are different
+			if bytes.Equal(blockID1.Hash, blockID2.Hash) {
+				return true
+			}
```

Comment:
If both headers are at the same height then the hashes must be different in order to be valid fork misbehaviour.

```suggestion
			if !bytes.Equal(blockID1.Hash, blockID2.Hash) {
				return true
			}
```